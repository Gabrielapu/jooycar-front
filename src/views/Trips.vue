<template>
  <div class="mx-4 my-2">
    <h1 class="text-h6"> 
      Listado de viajes
      <h2 class="text-subtitle-1">Este es un listado de tus viajes</h2> 
    </h1>
  </div>
  <hr class="mx-4 mb-2">
  <Filters @search="e => fetchTrips(e)"/>
  <Table
    :loading="loading"
    :trips="trips"
  />
</template>

<script setup>
import Table from '../components/Table.vue'
import Filters from '../components/Filters.vue'
import { onMounted, reactive, ref } from 'vue';
import { useTripStore } from '../stores/trip.js';

const tStore = useTripStore()
const loading = ref(false)
let trips = reactive([])

onMounted(async () => {
  await fetchTrips()
})

async function fetchTrips(filters) {
  loading.value = true
  let payload = {};
  if(filters) {
    const { startDate, endDate, distance } = filters
    payload = {
      start: startDate ? Date.parse(startDate) : null,
      end: endDate ? Date.parse(endDate) : null,
      distance: distance !== null
        ? Math.abs(distance) 
        : null  
    }
  }
  await tStore.getTrips(payload)
  trips = tStore.trips
  loading.value = false
}
</script>

<style scoped> 
.filters {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 16px;
}
</style>