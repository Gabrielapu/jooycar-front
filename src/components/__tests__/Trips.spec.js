import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { createPinia } from 'pinia'
import Trips from '@/views/Trips.vue'


describe('Trips.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(Trips, {
      global: {
        plugins: [createPinia()]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})