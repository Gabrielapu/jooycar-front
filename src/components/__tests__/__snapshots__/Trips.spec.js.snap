// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filters.vue > renders correctly 1`] = `
"<div data-v-11dc0277="" class="mx-4 my-2">
  <h1 data-v-11dc0277="" class="text-h6"> Listado de viajes <h2 data-v-11dc0277="" class="text-subtitle-1">Este es un listado de tus viajes</h2>
  </h1>
</div>
<hr data-v-11dc0277="" class="mx-4 mb-2">
<div data-v-11dc0277="" class="filters">
  <v-menu>
    <v-date-picker header="" title="Seleccionar fecha"></v-date-picker>
  </v-menu>
  <v-menu>
    <v-date-picker header="" title="Seleccionar fecha"></v-date-picker>
  </v-menu>
  <v-text-field label="Distancia" class="mr-2" variant="outlined" density="compact" clearable="" hide-details="" type="number" min="0"></v-text-field>
  <v-btn color="primary" variant="outlined" class="mr-4"> Buscar </v-btn>
</div>
<v-table data-v-11dc0277="" class="elevation-1 mx-4 border-sm rounded-lg">
  <thead>
    <tr>
      <th class="text-body-1 text-center"> Inicio </th>
      <th class="text-body-1 text-center"> Término </th>
      <th class="text-body-1 text-center"> Distancia y tiempo </th>
      <th class="text-body-1 text-center"> Acciones </th>
    </tr>
  </thead>
  <!--v-if-->
  <tbody></tbody>
</v-table>"
`;

exports[`Trips.vue > renders correctly 1`] = `
"<div data-v-11dc0277="" class="mx-4 my-2">
  <h1 data-v-11dc0277="" class="text-h6"> Listado de viajes <h2 data-v-11dc0277="" class="text-subtitle-1">Este es un listado de tus viajes</h2>
  </h1>
</div>
<hr data-v-11dc0277="" class="mx-4 mb-2">
<div data-v-11dc0277="" class="filters">
  <v-menu>
    <v-date-picker header="" title="Seleccionar fecha"></v-date-picker>
  </v-menu>
  <v-menu>
    <v-date-picker header="" title="Seleccionar fecha"></v-date-picker>
  </v-menu>
  <v-text-field label="Distancia" class="mr-2" variant="outlined" density="compact" clearable="" hide-details="" type="number" min="0"></v-text-field>
  <v-btn color="primary" variant="outlined" class="mr-4"> Buscar </v-btn>
</div>
<v-table data-v-11dc0277="" class="elevation-1 mx-4 border-sm rounded-lg">
  <thead>
    <tr>
      <th class="text-body-1 text-center"> Inicio </th>
      <th class="text-body-1 text-center"> Término </th>
      <th class="text-body-1 text-center"> Distancia y tiempo </th>
      <th class="text-body-1 text-center"> Acciones </th>
    </tr>
  </thead>
  <!--v-if-->
  <tbody></tbody>
</v-table>"
`;
